language/move-lang/src/hlir/ast.rs

```diff
@@ -398,15 +398,12 @@ impl BaseType_ {
         sp(loc, BaseType_::Apply(kind, n, ty_args))
     }
 
-    pub fn abilities(&self) -> AbilitySet {
+    pub fn abilities(&self, loc: Loc) -> AbilitySet {
         match self {
             BaseType_::Apply(abilities, _, _) | BaseType_::Param(TParam { abilities, .. }) => {
                 abilities.clone()
             }
-            BaseType_::Unreachable | BaseType_::UnresolvedError => panic!(
-                "ICE unreachable/unresolved error has no kind. Should only exist in dead code \
-                 that should not be analyzed"
-            ),
```

Comment:
I didn't quite see why this `panic!` is replaced with `AbilitySet::all`, maybe some comments?

Reply:
It was just missed from the refactor. It shouldn't really happen but the API was just wrong